File: Client Applications/Ruby Client/work_search.rb
require './util'
require './settings'
require 'open-uri'
require 'rubygems'
require 'json'

def beginWorkSearch(workName, applicationID, applicationKey)

	# Set the request URL
	url = $apiAddress + 'Works/?name=' + URI::encode(workName) + '&format=json'
		
	# Send the GET request
	resp = request(url, applicationID, applicationKey)

	# Interpret the JSON response 
	data = JSON.parse(resp.body)

	# Return the collection of track objects
	return data['ResultSet']
	
end

# Search for works by name
puts 'Please enter a work name:'
workName = gets.chomp
result = beginWorkSearch(workName, $applicationID, $applicationKey)

# Output the search results
result.each { 
	|work| puts work['Name'] + ' - ' + work['Composers']
}

puts 'Press Enter to continue...'
input = gets.chomp

File: Client Applications/Python Client/work_search.py
import util
import json
import urllib
import settings

def beginWorkSearch(workName, applicationID, applicationKey):

	# Set the request URL
	url = settings.apiAddress + 'Works/?name=' + urllib.parse.quote_plus(workName) + '&format=json'

	# Send the GET request
	resp = util.request(url, applicationID, applicationKey)

	# Interpret the JSON response 
	data = json.loads(resp.decode('utf8'))

	# Return the collection of track objects
	return data['ResultSet']

# Search for tracks by name	
workName = input('Please enter a work name:')
result = beginWorkSearch(workName, settings.applicationID, settings.applicationKey)

# Output the search results
for work in result:
	print(work['Name'] + ' - ' + work['Composers'])

input('Press Enter to continue...')